batch/batches/LiveToVOD/KAsyncLiveToVod.class.php

```diff
@@ -0,0 +1,240 @@
+<?php
+/**
+ * @package Scheduler
+ * @subpackage Copy
+ */
+
+/**
+ * Will copy objects and add them
+ * according to the suppolied engine type and filter 
+ *
+ * @package Scheduler
+ * @subpackage Copy
+ */
+class KAsyncLiveToVod extends KJobHandlerWorker
+{
+	/* (non-PHPdoc)
+	 * @see KBatchBase::getType()
+	 */
+	public static function getType()
+	{
+		return KalturaBatchJobType::LIVE_TO_VOD;
+	}
+	/**
+	 * (non-PHPdoc)
+	 * @see KBatchBase::getJobType()
+	 */
+	protected function getJobType()
+	{
+		return KalturaBatchJobType::LIVE_TO_VOD;
+	}
+	
+	/* (non-PHPdoc)
+	 * @see KJobHandlerWorker::exec()
+	 */
+	protected function exec(KalturaBatchJob $job)
+	{
+		return $this->copyCuePoint($job, $job->data);
+	}
+	
+	/**
+	 * Will take a data and copy cue points
+	 */
+	private function copyCuePoint(KalturaBatchJob $job, KalturaLiveToVodJobData $data)
+	{
+		$amfArray = json_decode($data->amfArray);
+		$currentSegmentStartTime = self::getSegmentStartTime($amfArray);
+		$currentSegmentEndTime = self::getSegmentEndTime($amfArray, $data->lastSegmentDuration);
+		self::normalizeAMFTimes($amfArray, $data->totalVODDuration, $data->lastSegmentDuration);
+
+		$totalCount = self::getCuePointCount($data->liveEntryId, $currentSegmentEndTime);
+		KalturaLog::info("Total count of cue-point to copy: " .$totalCount);
+		if ($totalCount == 0)
+			return $this->closeJob($job, null, null, "No cue point to copy", KalturaBatchJobStatus::FINISHED);
+
+		do
+		{
+			$copiedCuePointIds = array();
+			$liveCuePointsToCopy = self::getCuePointlistForEntry($data->liveEntryId, $currentSegmentEndTime);
+			if (count($liveCuePointsToCopy) == 0)
+				break;
+
+			//set the parnter ID for adding the new cue points in multi request
+			KBatchBase::impersonate($liveCuePointsToCopy[0]->partnerId);
+			KBatchBase::$kClient->startMultiRequest();
+			foreach ($liveCuePointsToCopy as $liveCuePoint)
+			{
+				$copiedCuePointId = self::copyCuePointToVOD($liveCuePoint, $currentSegmentStartTime, $amfArray, $data->vodEntryId);
+				if ($copiedCuePointId)
+					$copiedCuePointIds[] = $copiedCuePointId;
+			}
+			KBatchBase::$kClient->doMultiRequest();
+			KBatchBase::unimpersonate();
+			
+			//start post-process for all copied cue-point
+			KalturaLog::info("Copied [".count($copiedCuePointIds)."] cue-points");
+			self::postProcessCuePoints($copiedCuePointIds);
+
+			//decrease the totalCount (as the number of cue point return from server)
+			$totalCount -= count($liveCuePointsToCopy);
+		} while ($totalCount);
+
+		return $this->closeJob($job, null, null, "Copy all cue points finished", KalturaBatchJobStatus::FINISHED);
+	}
+
+
+	
+	private static function postProcessCuePoints($copiedCuePointIds)
+	{
+		KBatchBase::$kClient->startMultiRequest();
+		foreach ($copiedCuePointIds as $copiedLiveCuePointId)
+			KBatchBase::$kClient->cuePoint->updateStatus($copiedLiveCuePointId, KalturaCuePointStatus::HANDLED);
+		KBatchBase::$kClient->doMultiRequest();
```

Comment:
Need to validate the multi response.
1. Add error to the log
2. Raise exception that will mark the batch as failure.